### ISSUE TITLE
Build problems

### ISSUE THREAD

**Issue by SolidEva** [SolidHal/icecat-arm#1]
Some known build issues with armhf, and the associated fixes:
Original bug report for arm64
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=825355
Followup for armhf:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=854640

Commits that fix the armhf build problem:
https://salsa.debian.org/mozilla-team/firefox/commit/09b47055110aa23217bde99e809844771be07d98
https://salsa.debian.org/mozilla-team/firefox/commit/2038b037a9cdb2661e14be8b3bb16e76dab0cc10#8756c63497c8dc39f7773438edf53b220c773f67

basically, we need gcc and the c++ compiler to have these flags:

-fno-schedule-insns
-fno-schedule-insns2

and probably these:

-fno-lifetime-dse -fno-delete-null-pointer-checks

-fno-lifetime-dse seems to be included already

These can be added in config.status, but get reset by ./configure

**Comment by SolidEva** [comment]
Located in build/icecat/icecat-52.6.0/build/moz.configure/
The file toochain.configure may be what we want to modify:


```
    # C99 compiles as e.g. C11 (as of writing, this is true of libnestegg, for
    # example)
    if info.language == 'C' and info.language_version != 199901:
        if info.type in ('clang-cl', 'clang', 'gcc'):
            append_flag('-std=gnu99')

    # Note: MSVC, while supporting C++11, still reports 199711L for __cplusplus.
    # Note: this is a strict version check because we used to always add
    # -std=gnu++11.
    if info.language == 'C++':
        if info.type in ('clang', 'gcc') and info.language_version != 201103:
            append_flag('-std=gnu++11')
        # MSVC 2015 headers include C++14 features, but don't guard them
        # with appropriate checks.
        if info.type == 'clang-cl' and info.language_version != 201402:
            append_flag('-std=c++14')
```

Adding the following to each append_flag:
`-fno-lifetime-dse -fno-delete-null-pointer-checks -fno-schedule-insns -fno-schedule-insns2`

**Comment by SolidEva** [comment]
Changing the lines to
```
    if info.language == 'C' and info.language_version != 199901:
        if info.type in ('clang-cl', 'clang', 'gcc'):
            append_flag('-std=gnu99')
            append_flag('-fno-lifetime-dse')
            append_flag('-fno-delete-null-pointer-checks')
            append_flag('-fno-schedule-insns')
            append_flag('-fno-schedule-insns2')


    # Note: MSVC, while supporting C++11, still reports 199711L for __cplusplus.
    # Note: this is a strict version check because we used to always add
    # -std=gnu++11.
    if info.language == 'C++':
        if info.type in ('clang', 'gcc') and info.language_version != 201103:
            append_flag('-std=gnu++11')
            append_flag('-fno-lifetime-dse')
            append_flag('-fno-delete-null-pointer-checks')
            append_flag('-fno-schedule-insns')
            append_flag('-fno-schedule-insns2')

        # MSVC 2015 headers include C++14 features, but don't guard them
        # with appropriate checks.
        if info.type == 'clang-cl' and info.language_version != 201402:
            append_flag('-std=c++14')
            append_flag('-fno-lifetime-dse')
            append_flag('-fno-delete-null-pointer-checks')
            append_flag('-fno-schedule-insns')
            append_flag('-fno-schedule-insns2')
```
Sets obj/config.status correctly

**Comment by SolidEva** [comment]
Resolved with a patch!
